backend/data-access/instructorsData.js

```diff
@@ -0,0 +1,84 @@
+const fs = require('fs');
+const PATH = './data/instructors.json';
+
+/**
+ * Get all instructors from database
+ * 
+ * @returns {Array} a list of all instructors stored in BD
+ */
+const getInstructors = () => {
+    return JSON.parse(fs.readFileSync(PATH));
+}
+
+/**
+ * Get an instructor with given id from database
+ * 
+ * @param {string} id 
+ * 
+ * @returns {object | null} instructor with given id, null if not found
+ */
+const getInstructorById = (id) => {
+    const instructors = getInstructors();
+    const instructor = instructors.find(instructor => instructor.id.$oid === id);
+    return instructor ? instructor : null;
+}
+
+/**
+ * Add an instructor to the database
+ * 
+ * @param {string} id 
+ * 
+ * @returns {object} instructor has been added to the db
+ */
+const addInstructor = (instructor) => {
+    const instructors = getInstructors();
+    instructors.push(instructor);
```

Comment:
Do the validation in the service here. Try to avoid putting logic (like validation) in the routes file. The routes file should ideally only call service methods.